forum/users/views.py

```diff
@@ -1,11 +1,30 @@
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
 from rest_framework_simplejwt.views import (
     TokenObtainPairView as BaseTokenObtainPairView,
-    TokenRefreshView as BaseTokenRefreshView
+    TokenRefreshView as BaseTokenRefreshView, TokenBlacklistView
 )
+from users.serializers import LogoutSerializer
+
 
 class TokenObtainPairView(BaseTokenObtainPairView):
     throttle_scope = 'token_obtain'
 
 
 class TokenRefreshView(BaseTokenRefreshView):
     throttle_scope = 'token_refresh'
+
+
+class LogoutAndBlacklistRefreshTokenView(APIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = LogoutSerializer
+
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        serializer.save()
```

Comment:
I think you could do without serializer, because after the user is logged in, `access` and `refresh` tokens are placed in the user's cookie, and when the user logs out, you can get the refresh token from request.COOKIES and blacklist it already in the view, and add try except there. Or you can leave it so, this variant is also appropriate.